redwood-runtime/src/commonMain/kotlin/app/cash/redwood/ui/OnBackPressedCallback.kt

```diff
@@ -17,6 +17,12 @@ package app.cash.redwood.ui
 
 public abstract class OnBackPressedCallback(enabled: Boolean) {
   public var isEnabled: Boolean = enabled
+    set(value) {
+      field = value
+      enabledChangedCallback?.invoke()
+    }
+
+  public var enabledChangedCallback: (() -> Unit)? = null
```

Comment:
This mechanism was inspired by Android's implementation:

https://github.com/androidx/androidx/blob/dd1fc315eabedbcd024dd5ae219961c7236031a1/activity/activity/src/main/java/androidx/activity/OnBackPressedCallback.kt#L51-L60